components/team_controller/actions/index.ts

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {Team} from '@mattermost/types/teams';
+import {ServerError} from '@mattermost/types/errors';
+
+import {ActionFunc} from 'mattermost-redux/types/actions';
+import {ChannelTypes} from 'mattermost-redux/action_types';
+import {getTeamByName, selectTeam} from 'mattermost-redux/actions/teams';
+import {forceLogoutIfNecessary} from 'mattermost-redux/actions/helpers';
+import {fetchMyChannelsAndMembers} from 'mattermost-redux/actions/channels';
+import {getGroups, getAllGroupsAssociatedToChannelsInTeam, getAllGroupsAssociatedToTeam, getGroupsByUserIdPaginated} from 'mattermost-redux/actions/groups';
+import {logError} from 'mattermost-redux/actions/errors';
+import {isCustomGroupsEnabled} from 'mattermost-redux/selectors/entities/preferences';
+import {getCurrentUser} from 'mattermost-redux/selectors/entities/users';
+import {getLicense} from 'mattermost-redux/selectors/entities/general';
+import {isGuest} from 'mattermost-redux/utils/user_utils';
+
+import {isSuccess} from 'types/actions';
+
+import {loadStatusesForChannelAndSidebar} from 'actions/status_actions';
+import {addUserToTeam} from 'actions/team_actions';
+
+import LocalStorageStore from 'stores/local_storage_store';
+
+export function initializeTeam(team: Team): ActionFunc<Team, ServerError> {
+    return async (dispatch, getState) => {
+        const state = getState();
+
+        dispatch({type: ChannelTypes.CHANNELS_MEMBERS_CATEGORIES_REQUEST, data: null});
+
+        const currentUser = getCurrentUser(state);
+        LocalStorageStore.setPreviousTeamId(currentUser.id, team.id);
+
+        dispatch(selectTeam(team.id));
+
+        if (isGuest(currentUser.roles)) {
+            dispatch({type: ChannelTypes.CHANNELS_MEMBERS_CATEGORIES_FAILURE, error: null});
+        }
+
+        try {
+            await dispatch(fetchMyChannelsAndMembers(team.id));
+            dispatch({type: ChannelTypes.CHANNELS_MEMBERS_CATEGORIES_SUCCESS, data: null});
```

Comment:
this is the action we send for https://github.com/mattermost/mattermost-webapp/pull/11113#discussion_r967874257